library/ssl_cli.c

```diff
@@ -531,6 +531,46 @@ static int ssl_write_truncated_hmac_ext( mbedtls_ssl_context *ssl,
 }
 #endif /* MBEDTLS_SSL_TRUNCATED_HMAC */
 
+#if defined(MBEDTLS_SSL_RECORD_SIZE_LIMIT)
+static int ssl_write_record_size_limit( mbedtls_ssl_context *ssl,
+                                        unsigned char *buf,
+                                        const unsigned char *end,
+                                        size_t *olen )
+{
+    unsigned char *p = buf;
+
+    int limit = MBEDTLS_SSL_IN_CONTENT_LEN;
+    /* Set the limit a little lower than our actual buffer size to account for
+       unknown overhead.  */
+    limit -= 256;
+    limit *= 0.95;
+    limit &= ~255;
+
+    *olen = 0;
+
+    if( ssl->conf->record_size_limit == MBEDTLS_SSL_RECORD_SIZE_LIMIT_DISABLED )
```

Comment:
Why is this not at the top of the function?

Reply:
Part of the reason is that this is C90 and we can't declare limit anywhere except at the top of the block.  And the unconditional write to *olen is just following the pattern of other functions in this file.